code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/projects/ProjectMemberResourceManager.java

```diff
@@ -55,14 +55,23 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
         Policy policy = (Policy) ApiContext.getContext().getPolicy();
         String id = RequestUtils.makeSingularStringIfCan(criteria.get("id"));
         if (StringUtils.isNotEmpty(id)) {
-            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            ProjectMember projectMember;
+            try {
+                 projectMember = authDao.getProjectMember(Long.valueOf(id));
+            } catch (NumberFormatException e) {
+                return null;
```

Comment:
I think you just said that this will 404 higher up anyway. Is that right? Why not also raise Client visible here for consistency?